frontend/src/core/entities/reducer.js

```diff
@@ -88,6 +96,12 @@ export default function reducer(
                     action.translation,
                 ),
             };
+        case RECEIVE_SIBLINGS:
+            return {
+                ...state,
+                preceding: state.entities.concat(action.entities.preceding),
+                succeeding: state.entities.concat(action.entities.succeeding),
```

Comment:
We shouldn't extend the `EntitiesState` (not even above in the initial state). Instead, we should extend the `entities` key with the new entities.

You could create a new function that will inject preceding entities before and succeeding entities after the given entity within the entities list. You could call it like this:
```js
entities: injectSiblingEntities(state.entities, action.entities, entity),
```

Reply:
I'm not too sure what or how I would use `entity` in this function

Reply:
You'd need to pass it here first from the `getSiblingEntities` action.

It's the selected entity - the one the user clicks on - and you'll need it to determine where to insert surrounding entities.